Makefile

```diff
@@ -22,11 +37,61 @@ grpc:
 			./proto/*.proto
 	@sed -i 's/^import.*_pb2/from . \0/' ./$(MODULE)/generated/*.py
 
+## run: Run the gateway server
+.PHONY: run
 run:
 	@python3 -m $(MODULE) -c $(CONFIG)
 
+## test: Run tests
+.PHONY: test
 test:
 	@pytest
 
+## image: Build spdk image if it does not exist
+.PHONY: spdk
+spdk:
+ifneq ($(DOCKER_VERSION), $(SPDK_VERSION))
+	docker build \
+	--network=host \
+	--build-arg spdk_version=$(SPDK_VERSION) \
+	--build-arg spdk_branch=ceph-nvmeof \
+	${DOCKER_NO_CACHE} \
+	-t spdk:$(SPDK_VERSION) -f docker/Dockerfile.spdk .
+else
+	@echo "Docker image for version: $(SPDK_VERSION) exists"
+endif
+
+## spdk_rpms: Copy the rpms from spdk container in output directory
+.PHONY: spdk_rpms
+spdk_rpms:
+	docker run --rm -v $(curr_dir)/output:/output spdk:$(SPDK_VERSION) \
+	bash -c "cp -f /tmp/rpms/*.rpm /output/"
+
+
+## gateway: Build the nvme gateway image. The spdk image needs to be built first.
+.PHONY: gateway
+gateway: spdk grpc
+	docker build \
+	--network=host \
+	${DOCKER_NO_CACHE} \
+	--build-arg spdk_version=$(SPDK_VERSION) \
+	-t ${CONT_NAME}:${CONT_VERS} -f docker/Dockerfile.gateway .
+
+## push_gateway: Publish container into the docker registry for devs
+.PHONY: push_gateway
+push_gateway: gateway
```

Comment:
... and `push-gateway-image` here?  These are just suggestions though, feel free to ignore.